File: Math/62.Unique_Paths/Unique_Paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [[0 for i in range(m)] for j in range(n)]
        dp[0][0] = 1
        for i in range(len(dp)):
            for j in range(len(dp[i])):
                if i > 0: dp[i][j] += dp[i - 1][j]
                if j > 0: dp[i][j] += dp[i][j - 1]
        return dp[-1][-1]

File: Math/62.Unique_Paths/Unique_Paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<vector<int>>dp(m,vector<int>(n));
        dp[0][0] = 1;
        for(int i = 0; i < dp.size(); i++){
            for(int j = 0; j < dp[i].size(); j++){
                if(i > 0){
                    dp[i][j] += dp[i - 1][j];
                }
                if(j > 0){
                    dp[i][j] += dp[i][j - 1];
                }
            }
        }
        return dp[m-1][n-1];
    }
};